Packages/com.unity.inputsystem/InputSystem/Actions/InputActionProperty.cs

```diff
@@ -157,8 +157,8 @@ public override int GetHashCode()
             return !left.Equals(right);
         }
 
-        [SerializeField] private bool m_UseReference;
-        [SerializeField] private InputAction m_Action;
-        [SerializeField] private InputActionReference m_Reference;
+        [SerializeField] internal bool m_UseReference;
```

Comment:
Would testing `InputActionProperty.action` for `null` rather than exposing these fields have the same result in `TrackedPoseDriver`?

It's implementation is:
` public InputAction action => m_UseReference ? m_Reference != null ? m_Reference.action : null : m_Action;`
which I think will return `null` in all cases `m_Reference` and `m_Action` will both be null.

Otherwise perhaps adding a new member here rather than exposing the private fields so that the `TrackedPoseDriver` code is not so dependent on how this is implemented.

Reply:
It wouldn't be exactly the same. If a user clicked Use Reference to toggle the value in the Inspector window, the `action` property might start evaluating to `null` but one of its fields is already set. I wanted the migration code in TPD to skip copying the value from the old serialized field if _any_ `InputActionProperty` value had already been created.

Instead of making the fields `internal`, I could expose all the serialized fields in new `public` properties, which goes along with the Unity coding style rules that all Inspector window values are accessible through API, but that would be an API addition and thus this PR couldn't be included in a patch release.

Reply:
Ah right, I get the logic now.
There is another PR incoming that looks like it will need a minor bump, so now might be a good opportunity for getting something in (https://github.com/Unity-Technologies/InputSystem/pull/1701).

`m_Reference` and `m_Action` have public properties already though and adding alternative versions of those that don't hold the invariants might be confusing. I don't know if exposing just `m_UseReference` will be enough for you to work with?

Alternatively perhaps another method like `IsEmpty` or `IsNull` to make the code in TrackedPoseDriver simple.

Reply:
Even if I added a property for `m_UseReference`, I wouldn't be able to duplicate the logic exactly.

To try to address this feedback, I added new `internal` getter properties for the `m_Action` and `m_Reference` fields and reverted the fields back to `private`, and updated TPD to use those instead.

In the future, we can make them public and potentially add setters along with a property for `m_UseReference`, but we were scheduling to include this bug fix in a patch release so we can't add `public` properties yet.